### PR TITLE
cmd/k8s-operator,k8s-operator: optionally configure Pod affinity rules for proxies

### LINKED ISSUE #11861: FR: Allow setting affinity rules on k8s operator proxyclass
### What are you trying to do?

Ensuring statefulsets are scheduled on different nodes would be helpful to ensure NICS don't get saturated

### How should we solve this?

Add affinity rules to the proxyclass

### What is the impact of not solving this?

Customers will end up with poor networking performance

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jaxxstorm**
Fixes #11861

### CODE REVIEW

**Comment by irbekrm** [comment]
Thanks @jaxxstorm !

A couple nits:
- sign the commit https://github.com/tailscale/tailscale?tab=readme-ov-file#contributing (that will fix the DCO test failure)
- add `Updates#11861` to your commit message (will fix the issuerbot failure)
- rename the commit message first line + the PR header to something like `cmd/k8s-operator,k8s-operator: optionally configure Pod affinity rules for proxies`

File: scripts/kube-deepcopy.sh [hunk 1]
```diff
@@ -8,4 +8,4 @@ set -eu
 # files. We want to exclude all kube-related code from plan9 builds because some
 # apimachinery libraries refer to syscalls that are not available for plan9
 # https://github.com/kubernetes/apimachinery/blob/v0.28.2/pkg/util/net/util.go#L42-L63 
-sed -i "1 s|$| \&\& \!plan9|" k8s-operator/apis/v1alpha1/zz_generated.deepcopy.go
+sed -i.bak "1 s|$| \\&\\& \\!plan9|" k8s-operator/apis/v1alpha1/zz_generated.deepcopy.go && rm k8s-operator/apis/v1alpha1/zz_generated.deepcopy.go.bak
```
Comment (irbekrm) [inline scripts/kube-deepcopy.sh:11]: Why this change? Did the command error on your machine?

Comment (jaxxstorm) [reply scripts/kube-deepcopy.sh:11]: `sed -i` requires an extension on FreeBSD/OS X mac, this makes it work for both Linux and OS X. If I run this on mac right now with the default `sed` I get:

```
sed: 1: "k8s-operator/apis/v1alp ...": invalid command code k
```

I can switch to GNU Sed on my machine, but figure this helps eveyrone

Comment (irbekrm) [reply scripts/kube-deepcopy.sh:11]: Makes sense 👍🏼 I assumed it was a Mac thing, but wanted to clarify

**Review by irbekrm — APPROVED** [review]
Thanks @jaxxstorm !

**Comment by kenlasko** [comment]
Should the Operator Helm chart be updated to support this as well?

**Comment by irbekrm** [comment]
The affinity for the operator itself can already be configured via the Helm chart https://github.com/tailscale/tailscale/blob/main/cmd/k8s-operator/deploy/chart/values.yaml#L43

We won't be adding more configuration options for proxies in the chart (in favour of `ProxyClass`).

**Comment by kenlasko** [comment]
I've already used `affinity` for the operator in the helm chart, but it doesn't appear to apply to proxies.
